modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeVelocityIfStatementsMigrationCheck.java

```diff
@@ -0,0 +1,165 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.CharPool;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.util.Stack;
+
+/**
+ * @author Nícolas Moura
+ */
+public class UpgradeVelocityIfStatementsMigrationCheck
+	extends BaseUpgradeVelocityMigrationCheck {
+
+	@Override
+	protected String migrateContent(String content) {
+		String[] lines = content.split(StringPool.NEW_LINE);
+
+		for (int i = 0; i < lines.length; i++) {
+			String line = lines[i];
+
+			String newLine = line;
+
+			boolean parenthesis = false;
+
+			if (line.contains(_VELOCITY_IF_START) &&
+				_isVelocityStatement(line, _VELOCITY_IF_START)) {
+
+				parenthesis = true;
+
+				newLine = StringUtil.replace(
+					newLine, _VELOCITY_IF_START, "<#if");
+
+				_replaceIfStatementEnd(lines, i);
+			}
+			else if (line.contains(_VELOCITY_ELSEIF_START) &&
+					 _isVelocityStatement(line, _VELOCITY_ELSEIF_START)) {
+
+				parenthesis = true;
+
+				newLine = StringUtil.replace(
+					newLine, _VELOCITY_ELSEIF_START, "<#elseif");
+			}
+			else if (line.contains(_VELOCITY_ELSE_START) &&
+					 !line.contains(_VELOCITY_ELSEIF_START) &&
+					 _isVelocityStatement(line, _VELOCITY_ELSE_START)) {
+
+				newLine = StringUtil.replace(
+					newLine, _VELOCITY_ELSE_START, "<#else>");
+			}
+
+			if (parenthesis) {
+				if (_hasBreakLine(newLine)) {
+					String nextLine = lines[i + 1];
+
+					lines[i + 1] = StringUtil.replaceLast(
+						nextLine, CharPool.CLOSE_PARENTHESIS,
+						CharPool.GREATER_THAN);
+				}
+				else {
+					newLine = StringUtil.replaceLast(
+						newLine, CharPool.CLOSE_PARENTHESIS,
+						CharPool.GREATER_THAN);
+				}
+
+				newLine = StringUtil.removeFirst(
+					newLine, StringPool.OPEN_PARENTHESIS);
+			}
+
+			lines[i] = newLine;
+		}
+
+		return com.liferay.petra.string.StringUtil.merge(
+			lines, StringPool.NEW_LINE);
+	}
+
+	private static void _replaceIfStatementEnd(String[] lines, int i) {
+		Stack<String> stack = new Stack<>();
+
+		stack.push(_VELOCITY_IF_START);
+
+		int j = i;
+
+		while (!stack.empty()) {
+			j += 1;
+
+			String nextLine = lines[j];
+
+			if (nextLine.contains(_VELOCITY_IF_START)) {
+				stack.push(_VELOCITY_IF_START);
+			}
+
+			if (nextLine.contains(_VELOCITY_FOREACH_START)) {
+				stack.push(_VELOCITY_FOREACH_START);
+			}
+
+			if (nextLine.contains(_VELOCITY_MACRO_START)) {
+				stack.push(_VELOCITY_MACRO_START);
+			}
+
+			if (nextLine.contains(_VELOCITY_END)) {
+				stack.pop();
+			}
+
+			if (stack.empty()) {
+				lines[j] = StringUtil.replace(
+					nextLine, _VELOCITY_END, "</#if>");
+			}
+		}
+	}
+
+	private boolean _hasBreakLine(String newLine) {
+		boolean breakLine = false;
```

Comment:
@nicolas-sss, can you please save the declaration of this variable and use the statement `return`?